probe-rs/src/core/registers.rs

```diff
@@ -13,25 +17,86 @@ pub enum RegisterDataType {
     FloatingPoint,
 }
 
-/// Describes a register with its properties.
+/// This is used to label the register with a specific role that it plays during program execution and exception handling.
+/// This denotes the purpose of a register (e.g. `return address`),
+/// while the [`CoreRegister::name`] will contain the architecture specific label of the register.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub(crate) enum RegisterRole {
```

Comment:
This seems to miss return values, which are not exactly the same as the argument registers.

Reply:
I will fix it.